### DISCUSSION TITLE
Custom Function Instantiation

### DISCUSSION THREAD

**Discussion by vikashg** [Project-MONAI/model-zoo#89] [Q&A]
Hi everyone, 
I am trying to write a monai bundle for classification. As most of the examples are for segmentation, I figured I will have to write a custom function to load the image and the labels, which I wrote but it fails to instantiate 
This is the function 
```
class CreateImageLabelList():
   def __init__(self, filename):
       self.filename = filename

   def create_dataset(self):
       fid = open(self.filename, 'r')
       json_dict = json.load(fid)
       image_list = []
       label_list = []

       training_list = json_dict['Train']
       validation_list = json_dict['Validation']
       test_list = json_dict['Test']

       for _ in training_list:
           image_list.append(_['image'])
           label_list.append(_['label'])

       return image_list, label_list
```

This is the relevant part of the train.json script

```
import": ["$import glob", "$import os",
  "$from model_def import ModelDefinition",
  "$from monai.losses import DiceCELoss",
  "$from torch.nn import CrossEntropyLoss",
  "$from scripts.createList import CreateImageLabelList"],
"data": {
  "_target_": "CreateImageLabelList"
},
```

This is the error I get 

```
  File "/raid/Vikash/Tools/MONAI-DEPLOY/MONAI/monai/bundle/config_item.py", line 287, in instantiate
    raise RuntimeError(f"Failed to instantiate {self}.") from e
RuntimeError: Failed to instantiate {'_target_': 'CreateImageLabelList'}.
```

I know I am doing something wrong here. But I dont know what it is :( 
Any advice ?

Thanks 
Vikash

**Comment by Nic-Ma** [comment]
Hi @vikashg ,

You need to provide the full path for non-monai components, please refer to this example:
https://github.com/Project-MONAI/tutorials/tree/main/modules/bundle/custom_component

Thanks.

**Reply by vikashg** [reply]
Hey @Nic-Ma, 
Thanks for the comments, I am trying to build the model. I think I have gotten it to a point where I can share some code for review and ask more pointed questions. There is a branch vikash/breastclassification. If you can go through it once it will be really helpful. I do not think that this section I wrote in the train.json is correct 

```
"key_metric": {
  "train_accuracy": {
     "_target_": "ignite.metrics.Accuracy",
     "output_transform": "$monai.handlers.from_engine(['pred', 'label'])"
  }
```
Because I am not using the dictionary transforms (this is deliberate attempt to break), so I do not think this is the right way to calculate accuracy. Further, in order to test the different sections of the config file I am using the function script.py and running it as 
```
python script.py
```
Please have a look and with some help I think I can complete this bundle. 
Thanks so much.

**Reply by Nic-Ma** [reply]
Hi @vikashg ,

Could you please help submit your bundle PR for review?
Then we can discuss more details in the code comments directly.

Thanks.

**Reply by vikashg** [reply]
Yes I will do that today. I didnt think it was ready for a PR. But oh well, its better to fix that way. 👍
